I-D_in_xml/draft-ietf-ccamp-optical-impairment-topology-yang.xml

```diff
@@ -1673,12 +1668,154 @@ Line |     .  +-----+ | |  *  *  | | +-----+  .     | Line
   </t>
   
   <t>
-   In case of individual OTSi protection, there are two end-to-end MC
-   paths associated with the OTSi signal. In the YANG model, this is
-   modeled as a leaf list of the otsi providing the e2e-mc-path-id for
-   the two end-to-end MC paths associated with the individually
-   protected otsi.
+   In case of individual OTSi protection without 3R regenerators,
```

Comment:
Much of the editing work is due to the discrepancy between the graphics depicting bidirectional paths but the text needs to refer to them as unidirectional. For example, a bidirectional arrow cannot represent a single path, OMS or TE-link but two. 
The Group has decided in the resolution for #132 that the protection graphs shall be depicted as bidirectional. Therefore the text has to be adapted to reflect that the bidirectional arrow represents in fact two path: up- and downstream. This causes quite some changes in the text. 
If instead #132 would be reversed, the graphs could be depicted as unidirectional and the text would remain largely intact and easier to read. For comparison here an edited version with unidirectional graphs and corresponding text:
 ```suggestion
In case of individual OTSi protection without 3R regenerators,
   two end-to-end MC paths are associated with the OTSi signal. In the
   YANG model, this is modeled as leaf list of the otsi providing the
   e2e-mc-path-id for the two end-to-end MC paths associated with the
   individually 1+1 protected OTSi. This scenario is depicted in
   <xref target="OTSi_prot_no_3R"/> below.
  </t>


   <figure align="center" title="Individual OTSi Protection without 3R
   regenerators" anchor="OTSi_prot_no_3R">
    <artwork><![CDATA[
                         end-to-end MC path1
      |------------------------------------------------------->|
 +-----------+                                          +-----------+
 | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
 |      +----|           | WDM |      | WDM |           |----+      |
 |      |   -o---------->o-----o----->o-----o---------->o-   |      |
 |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
 | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
-o-o  o-o-   |                                          |   -o-o  o-o-
 | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
 |      |  \ |     | WDM |     | WDM  |     | WDM |     | /  |      |
 |      |   -o---->o-----o---->o------o---->o-----o---->o-   |      |
 |      +----|     |Node5|     | Node6|     |Node7|     |----+      |
 |           |     +-----+     +------+     +-----+     |           |
 +-----------+                                          +-----------+
      |------------------------------------------------------>|
                         end-to-end MC path2

]]> </artwork>
  </figure>

  <t>
   For each OMS MCG (TE-link) along the two end-to-end MC paths, the
   e2e-mc-path-id is provided for the individually protected OTSi
   signal. Based on this information, it is possible to construct the
   two end-to-end MC paths between the optical transponders terminating
   the individually 1+1 protected OTSi.
  </t>

  <t>
   In the scenario depicted in <xref target="OTSi_prot_no_3R"/>, the
   e2e-mc-path-id of end-to-end MC path1 is provided for the  TE-links
   between WDM Node1 and WDM Node3, WDM Node3 and WDM Node4 as well as
   WDM Node4 and WDM Node2 while the e2e-mc-path-id of end-to-end MC
   path2 is provided for the TE-links between WDM Node1 and WDM Node5,
   WDM Node5 and WDM Node6, WDM Node6 and WDM Node7 as well as WDM
   Node7 and WDM Node2.
  </t>

  <t>
   If a 3R regenerator is crossed on one of the two legs or even on
   both legs, the end-to-end MCs are terminated on both sides of the
   3R regenerator. The configured-termination-type attribute set to
   "3r-regeneration" shall be used to indicate that the transceivers
   are forming a 3R regenerator instead of terminating the layer 0
   tunnel (layer 0 service).
   At WDM-nodes containing a 3R regenerator, the end-2-end MCs are
   stitched together forming the end-to-end path for the layer 0 tunnel
   (layer 0 service). This is reflected in the leaf list of the OTSi,
   which now lists all e2e-mc-path-ids of the end-to-end MC paths on
   the two legs of the individually 1+1 protected OTSi signal.
  </t>

  <t>
   In the scenario depicted in <xref target="OTSi_prot_with_3R"/> below
   where a 3R regenerator is crossed in WDM Node6 on the lower leg,
   the e2e-mc-path-id leaf list has 3 entries:

   <ol>
     <li>
       The e2e-mc-path-id identifying end-to-end MC path1 from WDM
       Node1 via WDM Node3 and WDM Node4 to WDM Node2 (upper leg)
     </li>
     <li>
       The e2e-mc-path-id identifying end-to-end MC path2 from WDM
       Node1 via WDM Node5 to WDM Node6 containing the 3R regenerator
       (left hand segment of the lower leg)
     </li>
     <li>
       The e2e-mc-path-id identifying end-to-end MC path3 from WDM
       Node6 containing the 3R regenerator via WDM Node7 to WDM
       Node2 (right hand segment of the lower leg)
     </li>
   </ol>

   Based on this modelling approach it is possible to identify the
   end-2-end MCs stitched together at 3R regenerators on each of the
   two legs of the individually protected 1+1 OTSi signal. Like for
   the case without 3R regenerators is also possible to associate
   two end-to-end paths for the two legs between the optical
   transponders terminating the individually 1+1 protected OTSi
   in WDM Node1 and WDM Node2, respectively:

   <ol spacing="compact">
     <li>end-to-end MC path1 (upper leg)</li>
     <li>end-to-end MC path2 + end-to-end MC path3 (lower leg)</li>
   </ol>

  </t>

  <figure align="center" title="Individual OTSi Protection with a 3R
   regenerator" anchor="OTSi_prot_with_3R">
    <artwork><![CDATA[
                         end-to-end MC path1
      |------------------------------------------------------->|
 +-----------+                                          +-----------+
 | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
 |      +----|           | WDM |      | WDM |           |----+      |
 |      |   -o---------->o-----o----->o-----o---------->o-   |      |
 |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
 | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
-o-o  o-o-   |                                          |   -o-o  o-o-
 | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
 |      |  \ |     |     |     | +--+ |     |     |     | /  |      |
 |      |   -o---->o-----o---->o-o  o-o---->o-----o---->o-   |      |
 |      +----|     | WDM |     | +--+ |     | WDM |     |----+      |
 |           |     |Node5|     |  3R  |     |Node7|     |           |
 +-----------+     +-----+     +------+     +-----+     +-----------+
                               WDM Node6
                               with 3R
                               Regenerator
      |------------------------->|  |------------------------->|
          end-to-end MC path2           end-to-end MC path3

]]> </artwork>
  </figure>

  <t>Individual OTSi protection use cases:</t>
    <ol type="(%i)">
@@ -2157,10 +2294,74 @@ Line |     .  +-----+ | |  *  *  | | +-----+  .     | Line
   feasibility consideration).
  </t>

  <figure align="center" title="Modeling view of 1+1 protected TE-link
   with underlying primary and secondary TE-link"
   anchor="OMS_MCG_prot_link_1">
    <artwork><![CDATA[
         WDM-TE-Node-1                         WDM-TE-Node-2
    +-----------------------+           +-----------------------+
    |     ROADM     Splitter|           |Selector     ROADM     |
    |   +-------+   +-------+LTP2   LTP4+-------+   +-------+   |
    |   |       |   |    ---o---------->o---    |   |       |   |
LTP1|   |    RP1|   |   /   |  prim.    |   \   |   |RP2    |   |LTP6
<-->o---o.......o---o--o    |           |    o--o---o.......o---o<-->
    |   |       |   |   \   |  second.  |   /   |   |       |   |
    |   |       |   |    ---o---------->o---    |   |       |   |
    |   +-------+   +-------+LTP3   LTP5+-------+   +-------+   |
    |                       |           |                       |
    +-----------------------+           +-----------------------+
              --+                                   +--
     ROADM port |                                   | ROADM port
            RP1 o---------------------------------->o RP2
                |                                   |
              --+                                   +--
                   1+1 protected OMS MCGs (TE-links)

                   between ROADM ports RP1 and RP2
             underlying primary and secondary TE-links:
                          --+           +--
                            |  prim.    |
                       LTP2 o---------->o LTP4
                       LTP3 o---------->o LTP5
                            |  second.  |
                          --+           +--
        connectivity matrix provides optical impairments in
        both directions between LTPs in the two WDM-TE-Nodes:
        * LTP1 and LTP2,                     * LTP6 and LTP4,
        * LTP1 and LTP3                      * LTP6 and LTP5
]]> </artwork>
  </figure>

  <t>
   [Editor's note: text and a figure based on the slide deck attached
   to Open Issue #119 on GitHub showing more details regarding the LTPs
   of the primary and secondary TE-link will be added here.]
   <xref target="OMS_MCG_prot_link_1" /> illustrates this modeling
   approach including the LTPs in WDM-TE-Node-1 and WDM-TE-Node-2,
   respectively.
   In addition to the physical view, the following three TE-links are
   shown:

   <ul spacing="compact">
     <li>The 1+1 protected TE-link</li>
   </ul>
   <ul spacing="compact">
     <li><t>The underlying primary TE-link</t></li>
     <li>The underlying secondary TE-link</li>
   </ul>

   The optical impairments of the splitter (outgoing direction) and
   the selector (incoming direction) are included in the optical
   impairments described by the connectivity matrix and the local link
   connectivity list for the TE node.
   For the example shown in <xref target="OMS_MCG_prot_link_1" />,
   the connectivity matrix describes the optical impairments between
   LPT1 and LTP2 as well as LTP1 and LTP3 for WDM-TE-Node-1. Likewise,
   the connectivity matrix describes the optical impairments between
   LPT4 and LTP6 as well as LTP5 and LTP6 in WDM-TE-Node-2.
  </t>

  </section>
@@ -2189,15 +2390,106 @@ Line |     .  +-----+ | |  *  *  | | +-----+  .     | Line

   The optical impairments exposed for this 1+1 protected TE-link are
   typically based on the optical impairments of the worse of the two
   underlying physical OMS links.
   underlying physical OMS links including the optical impairments
   imposed by the splitter (outgoing direction) and selector (incoming
   direction).
  </t>


  <t>
   <xref target="OMS_MCG_prot_link_2a"/> illustrates this modeling
   approach where the splitter/selector in the adjacent WDM-TE-Nodes,
   WDM-TE-Node-1 and WDM-TE-Node-2, as well as the two OMS MCG
   links are abstracted into a single 1+1 protected TE-link. This is
   illustrated by the the dotted line surrounding these four physical
   entities in <xref target="OMS_MCG_prot_link_2a"/>. Based on this
   modeling approach, the ROADM port connected to the splitter/selector
   function is modeled as LTP for the 1+1 protected TE-link (LTP2 in
   WDM-TE-Node-1 and LTP3 in WDM-TE-Node-2).
   In this example, the connectivity matrix describes the optical
   impairments between LPT1 and LTP2 in WDM-TE-Node-1. Likewise, the
   connectivity matrix describes the optical impairments between LPT3
   and LTP4 in WDM-TE-Node-2.
  </t>

  <figure align="center" title="Modeling view of abstracted 1+1 protected
   TE-link - ROADM ports modeled as LTPs" anchor="OMS_MCG_prot_link_2a">
    <artwork><![CDATA[
    
         WDM-TE-Node-1                         WDM-TE-Node-2
    +-----------------------+           +-----------------------+
    |     ROADM     Splitter|           |Selector     ROADM     |
    |   +-------+   +.......+...........+-------+   +-------+   |
    |   |       |   .    ---o---------->o---    .   |       |   |
LTP1|   |   LTP2|   .   /   |           |   \   .   |LTP3   |   |LTP4
<-->o---o.......o---o--o    |           |    o--o---o.......o---o<-->
    |   |       |   .   \   |           |   /   .   |       |   |
    |   |       |   .    ---o---------->o---    .   |       |   |
    |   +-------+   +...................+.......+   +-------+   |
    |                       |           |                       |
    +-----------------------+           +-----------------------+
              --+                                   +--
     ROADM port |                                   | ROADM port
           LTP2 o---------------------------------->o LTP3
                |                                   |
              --+                                   +---
                  Splitter/Selector abstracted into
                  1+1 protected OMS MCG (TE-link)

        connectivity matrix provides optical impairments in
        both directions between LTPs in the two WDM-TE-Nodes:
        * LTP1 and LTP2                      * LTP3 and LTP4

]]> </artwork>
  </figure>

  <t>
   [Editor's note: text and a figure based on the slide deck attached
   to Open Issue #119 on GitHub showing more details regarding the LTP
   of the 1+1 protected TE-link will be added here.]
   Alternatively, the optical impairments imposed by the splitter and
   selector in each of the two adjacent WDM-TE-Nodes can also be
   included in the optical impairments described by the connectivity
   matrix of the two nodes instead of taking them into account as
   optical impairments of the 1+1 protected TE-link.
   This is illustrated in <xref target="OMS_MCG_prot_link_2b"/> below.
   In this case, the two physical ports on both ends of the 1+1
   protected TE-link are abstracted into a single LTP, LTP2 and LTP3,
   respectively.
  </t>

  <figure align="center" title="Modeling view of abstracted 1+1 protected
   TE-link - physical ports abstracted into single LTP on both link ends"

   anchor="OMS_MCG_prot_link_2b">
    <artwork><![CDATA[
    
         WDM-TE-Node-1                         WDM-TE-Node-2
    +-----------------------+           +-----------------------+
    |     ROADM     Splitter|           |Selector     ROADM     |
    |   +-------+   +-------+...........+-------+   +-------+   |
    |   |       |   |    ---o---------->o---    |   |       |   |
LTP1|   |       |   |   /   |LTP2   LTP3|   \   |   |       |   |LTP4
<-->o---o.......o---o--o    |           |    o--o---o.......o---o<-->
    |   |       |   |   \   |LTP2   LTP3|   /   |   |       |   |
    |   |       |   |    ---o---------->o---    |   |       |   |
    |   +-------+   +-------+...........+-------+   +-------+   |
    |               Selector|           |Splitter               |
    +-----------------------+           +-----------------------+

                          --+           +--
                            |           |
                       LTP2 o---------->o LTP3
                            |           |
                          --+           +--
                            1+1 protected
                            OMS MCG (TE-link)

        connectivity matrix provides optical impairments in
        both directions between LTPs in the two WDM-TE-Nodes:
        * LTP1 and LTP2                      * LTP3 and LTP4
]]> </artwork>
  </figure>

```

Reply:
@sergiobelotti  not sure I got your question. The green part is simply a version with **unidirectional graphs** and corresponding text. It's just for illustration.
Does this clarify?

Reply:
> @sergiobelotti not sure I got your question. The green part is simply a version with **unidirectional graphs** and corresponding text. It's just for illustration. Does this clarify?

It's very simple what I ask you: it is clear there is a unidir graphs , I wan to see what you have modified in the text . Very simple. The greenpiece does not illustrate anything if it is not clear what in parallel of changing the figure you changed in the text. Hope this can clarify.

Reply:
Few things here: 
- respecting the decision to address #132, I already have provided a review of the text and aligned it with the bidirectional graphs. This is the "official" review part that needs to be addressed.
- the text you are referring to is a "what-if" text for **information only** and not a review. 
- it is challenging to provide a diff between a comment and a text in review that has already been reviewed. That said, I took the original text, put the graphs in shape, and removed the following blurb starting at line 1680 in the original:
`
  <t>
   Note: Where several WDM nodes are involved, the links between WDM
   nodes in the figures below illustrating  the various protection
   architectures are drawn as bi-directional links (arrows on both ends
   of the dashed line) for the sake of simplification.
   Physically, all links in WDM networks are uni-directional and there
   is a fiber link in forward and a fiber link in reverse direction.
   WDM services are typically bi-directional and the same media channel
   is used on the links in both directions.
   The splitter function of the combined splitter/selector is always
   located on the upstream side whereas the selector function is always
   located on the downstream side in the two directions.
  </t>
`

Reply:
Here is my suggestion how to address the comment regarding the bidirectional arrows used in some of the figures:

- remove the Note added based on the agreement in the call on 2023-06-20
- replace each of the "simplified" figures with 2 figures, one for each direction and align the text in the sections containing the figures accordingly (all other sections can be left unchanged).

Example: Figure 11 will be replaced by the following 2 figures:

```
                           end-to-end MC path1
        |------------------------------------------------------->|

   +-----------+                                          +-----------+
   | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
   |      +----|           | WDM |      | WDM |           |----+      |
   |      |   -o---------->o-----o----->o-----o---------->o-   |      |
   |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
   | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
  -o-o  o-o-   |                                          |   -o-o  o-o-
   | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
   |      |  \ |     | WDM |     | WDM  |     | WDM |     | /  |      |
   |      |   -o---->o-----o---->o------o---->o-----o---->o-   |      |
   |      +----|     |Node5|     | Node6|     |Node7|     |----+      |
   |   Splitter|     +-----+     +------+     +-----+     |Selector   |
   +-----------+                                          +-----------+

        |------------------------------------------------------->|
                           end-to-end MC path2


      Figure 11: Individual OTSi Protection without 3R regenerators
                           (forward direction)

                           end-to-end MC path1'
        |<-------------------------------------------------------|

   +-----------+                                          +-----------+
   | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
   |      +----|           | WDM |      | WDM |           |----+      |
   |      |   -o<----------o-----o<-----o-----o<----------o-   |      |
   |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
   | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
  -o-o  o-o-   |                                          |   -o-o  o-o-
   | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
   |      |  \ |     | WDM |     | WDM  |     | WDM |     | /  |      |
   |      |   -o<----o-----o<----o------o<----o-----o<----o-   |      |
   |      +----|     |Node5|     | Node6|     |Node7|     |----+      |
   |   Selector|     +-----+     +------+     +-----+     |Splitter   |
   +-----------+                                          +-----------+

        |<-------------------------------------------------------|
                           end-to-end MC path2'


      Figure 12: Individual OTSi Protection without 3R regenerators
                           (reverse direction)
```

Reply:
As a frequent user of standards, the minimum quality criteria for a standard document is that terminology is defined and the document is consistent. 
- new terminology is OK if it is defined in the doc or referenced from other standards
- If arrows go from left to right and/or right to left is not an issue, as long as the descriptive text is consistent with it. 
- Using one or two graphs is not an issue as long as the descriptive text is consistent with it.
I am happy to work on any resolution proposal that meets the quality criteria.

@sergiobelotti It would be great if other participants with similar quality expectations could provide a thorough review and share their comments to quickly close this debate.

Reply:
@ggrammel, could you please confirm whether you agree with what was suggested above (I need to know this because it will require some effort):

> -     remove the Note added based on the agreement in the call on 2023-06-20
> -     replace each of the "simplified" figures with 2 figures, one for each direction and align the text in the sections containing the figures accordingly (all other sections can be left unchanged).

A definition of "end-to-end MC path" is in progress based on what we discussed on 2023-06-27 and will be shared shortly.

It has always been the objective to meet the quality criteria for standards and we currently have 3 approvals (see for example https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/138#pullrequestreview-1500253687).

Reply:
Here is the text proposal for the end-to-end MC path:

```
   In section Section 2.10, the term "end-to-end MC path" is used to
   describe the topological aspect of the end-to-end MC, i.e., the path
   through the media (see: [G.807_Amd1], section 7.1.2).  This is in
   line with the TE path defined in [RFC8795], section 3.9, where the TE
   path is defined as "an ordered list of TE links and/or TE nodes on
   the TE topology graph" interconnecting a pair of tunnel termination
   points (TTPs).
```

Reply:
looks good!

Reply:
If the text is carefully updated and is carefully reviewed, there likelihood for inconsistencies is low.
Only a few paragraphs need to be updated. Below is an example. I will only proceed with the remaining paragraphs if there is agreement to go down that path.

The updated text right below new Figure 11 and Figure 12 would be the following:

```

                           end-to-end MC path1
        |------------------------------------------------------->|

   +-----------+                                          +-----------+
   | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
   |      +----|           | WDM |      | WDM |           |----+      |
   |      |   -o---------->o-----o----->o-----o---------->o-   |      |
   |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
   | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
  -o-o  o-o-   |                                          |   -o-o  o-o-
   | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
   |      |  \ |     | WDM |     | WDM  |     | WDM |     | /  |      |
   |      |   -o---->o-----o---->o------o---->o-----o---->o-   |      |
   |      +----|     |Node5|     | Node6|     |Node7|     |----+      |
   |   Splitter|     +-----+     +------+     +-----+     |Selector   |
   +-----------+                                          +-----------+

        |------------------------------------------------------->|
                           end-to-end MC path2


      Figure 11: Individual OTSi Protection without 3R regenerators
                           (forward direction)

                           end-to-end MC path1'
        |<-------------------------------------------------------|

   +-----------+                                          +-----------+
   | WDM Node1 |           +-----+      +-----+           | WDM Node2 |
   |      +----|           | WDM |      | WDM |           |----+      |
   |      |   -o<----------o-----o<-----o-----o<----------o-   |      |
   |  OT  |  / |           |Node3|      |Node4|           | \  |  OT  |
   | +--+ | /  |           +-----+      +-----+           |  \ | +--+ |
  -o-o  o-o-   |                                          |   -o-o  o-o-
   | +--+ | \  |     +-----+     +------+     +-----+     |  / | +--+ |
   |      |  \ |     | WDM |     | WDM  |     | WDM |     | /  |      |
   |      |   -o<----o-----o<----o------o<----o-----o<----o-   |      |
   |      +----|     |Node5|     | Node6|     |Node7|     |----+      |
   |   Selector|     +-----+     +------+     +-----+     |Splitter   |
   +-----------+                                          +-----------+

        |<-------------------------------------------------------|
                           end-to-end MC path2'


      Figure 12: Individual OTSi Protection without 3R regenerators
                           (reverse direction)

   For each OMS MCG (TE-link) along the two end-to-end MC paths in
   forward direction (end-to-end MC path1 and end-to-end MC path2) as
   well as the two end-to-end MC paths in reverse direction (end-to-end
   MC path1' and end-to-end MC path2'), the e2e-mc-path-id is provided
   for the individually protected OTSi signal.  Based on this
   information, it is possible to construct the end-to-end MC paths
   between the optical transponders terminating the individually 1+1
   protected OTSi.

   In the scenario depicted in Figure 11 and Figure 12, the e2e-mc-path-
   id of end-to-end MC path1 and end-to-end MC path1' is provided for
   the TE-links between WDM Node1 and WDM Node3, WDM Node3 and WDM Node4
   as well as WDM Node4 and WDM Node2 while the e2e-mc-path-id of end-
   to-end MC path2 and end-to-end MC path2' is provided for the TE-links
   between WDM Node1 and WDM Node5, WDM Node5 and WDM Node6, WDM Node6
   and WDM Node7 as well as WDM Node7 and WDM Node2.
```

Reply:
The snippet looks good to me. Appreciate the effort.

Reply:
@ggrammel , so I will proceed in the same way and will update the other figures as well as the related text.